lib/ResolverFactory.js

```diff
@@ -45,23 +72,33 @@ exports.createResolver = function(options) {
 	// changes the original options.plugins array, causing duplicate plugins
 	const plugins = (options.plugins && options.plugins.slice()) || [];
 
-	// A list of main fields in description files
-	let mainFields = options.mainFields || ["main"];
-
 	// A list of alias fields in description files
 	const aliasFields = options.aliasFields || [];
 
 	// A list of main files in directories
 	const mainFiles = options.mainFiles || ["index"];
 
-	// A list of extensions which should be tried for files
-	let extensions = options.extensions || [".js", ".json", ".node"];
+	/**
+	 * A list of extensions which should be tried for files
+	 * @type {Array<string>}
+	 */
+	const extensions = options.extensions
+		? [].concat(/** @type {Array<never>} */ (options.extensions))
+		: [".js", ".json", ".node"];
 
 	// Enforce that a extension from extensions must be used
 	const enforceExtension = options.enforceExtension || false;
 
-	// A list of module alias configurations or an object which maps key to value
-	let alias = options.alias || [];
+	/**
+	 * A list of module alias configurations or an object which maps key to value
+	 * @type {Array<AliasOptionEntry>}
+	 */
+	const alias =
+		typeof options.alias === "object" &&
+		!Array.isArray(options.alias) &&
+		options.alias !== null
+			? aliasOptionsToArray(options.alias)
+			: /** @type {Array<AliasOptionEntry>} */ (options.alias) || [];
 
 	// A PnP API that should be used - null is "never", undefined is "auto"
 	let pnpApi = options.pnpApi || null;
```

Comment:
it is pretty funny, but it actually impossible to create it in one ternary operator, since `@ts-ignore` must be placed on previous line, so on valid ts ignore comment prettier gives error
example:
```
condition
//@ts-ignore
? some_code
: else_statement
```
prettier error 👆

```
condition
? //@ts-ignore
  some_code
: else_statement
```
`@ts-ignore` does not work 👆 . Looks like it process next statement (else clause)

Reply:
I will create utility function in module scope that process pnp option

Reply:
done